src/Transport/HttpTransport.php

```diff
@@ -122,10 +129,17 @@ public function send(Event $event): ?string
      */
     private function cleanupPendingRequests(): void
     {
-        while ($promise = array_pop($this->pendingRequests)) {
+        while ($pendingRequest = array_pop($this->pendingRequests)) {
             try {
```

Comment:
This whole `try...catch` block could be extracted into a private method.

Reply:
True. Tried as little refactoring as possible, but can certainly extract this into a private method